pkg/controllers/hub/trafficmanagerbackend/controller.go

```diff
@@ -557,9 +569,9 @@ func equalAzureTrafficManagerEndpoint(current, desired armtrafficmanager.Endpoin
 		*current.Properties.EndpointStatus == *desired.Properties.EndpointStatus
 }
 
-// updateTrafficManagerEndpointsAndUpdateStatusIfUnknown updates the Azure Traffic Manager endpoints.
+// updateTrafficManagerEndpointsAndUpdateStatus updates the Azure Traffic Manager endpoints.
```

Comment:
the original func name is to indicate it will only update the status when the status is unknown. For the true/false condition, the caller needs to update the status